src/auth.js

```diff
@@ -0,0 +1,39 @@
+import * as jwt from 'jsonwebtoken'
+import { AuthorizationError } from 'graphql-auth-directives/dist/errors'
+import * as micromatch from 'micromatch'
+
+const JWT_SECRET = process.env.JWT_SECRET
+
+/**
+ * Verify if the request is authenticated and is authorized to access the given scope.
+ * @param context
+ * @param scope
+ * @returns {Object}
+ */
+export const verifyRequest = (context, scope) => {
+  const req = context.req || context.request
+  const token = req && (req.headers.authorization || req.headers.Authorization)
+
+  if (!token) {
+    throw new AuthorizationError({ message: 'No authorization token.' })
+  }
+
+  const id = token.replace('Bearer ', '')
+  let payload = null
+
+  try {
+    payload = jwt.verify(id, JWT_SECRET)
+  } catch (error) {
+    throw new AuthorizationError({
```

Comment:
just to confirm - is it normal for graphql responses to return http200 even when an error is being raised? This took me a while to debug in my python client.

Reply:
Yes, if the server is working and an error is expected, it will return the error as a GraphQL error.